packages/core/src/web/helpers/convertToPath.ts

```diff
@@ -54,34 +58,65 @@ export const convertTextToPath = async ({
   weldingTexts?: boolean;
 }): Promise<ConvertToPathResult> => {
   const isSubCommand = parentCommand !== undefined;
-  const isTextPath = element.getAttribute('data-textpath-g');
-  const textElem = isTextPath ? element.querySelector('text')! : element;
 
   if (textActions.isEditing) textActions.toSelectMode();
 
-  const res = await fontFuncs.convertTextToPath(textElem, { isSubCommand, weldingTexts });
+  const { command, path } = await fontFuncs.convertTextToPath(element, { isSubCommand: true, weldingTexts });
 
-  if (!res.path) {
-    return {
-      bbox: { height: 0, width: 0, x: 0, y: 0 } as DOMRect, // Default bbox if res is null
-      command: undefined,
-      path: undefined,
-    };
+  if (command && isSubCommand) {
+    parentCommand.addSubCommand(command);
   }
 
-  const { command, path, textPathPath } = res;
-
-  if (path && isToSelect) {
-    svgCanvas.multiSelect([path, textPathPath].filter(Boolean) as SVGElement[]);
+  if (command && !isSubCommand) {
+    undoManager.addCommandToHistory(command);
   }
 
-  if (command && isSubCommand) {
-    parentCommand.addSubCommand(command);
+  if (path && isToSelect) {
+    svgCanvas.selectOnly([path]);
   }
 
   return { bbox: path?.getBBox()!, command: parentCommand || command || undefined, path: path || undefined };
 };
 
+export const convertTextOnPathToPath = async ({
+  element,
+  isToSelect: _isToSelect,
+  parentCommand = new BatchCommand('Convert Text on Path to Path'),
+  weldingTexts = false,
+}: {
+  element: SVGElement;
+  isToSelect?: boolean;
+  parentCommand?: IBatchCommand;
+  weldingTexts?: boolean;
+}): Promise<ConvertToPathResult & { group: SVGGElement }> => {
+  const isSubCommand = parentCommand.getText() !== 'Convert Text on Path to Path';
+  const pathElement = element.querySelector('path');
+  const textElement = element.querySelector('text');
+
+  if (textActions.isEditing) textActions.toSelectMode();
+
+  svgCanvas.clearSelection();
+
+  const { command, path } = await fontFuncs.convertTextToPath(textElement!, { isSubCommand, weldingTexts });
```

Comment:
if you want to separate convertText and convertTextOnPath, please remove the `textPathPath` inside `font-func.ts`